modules/hostdb/scan.go

```diff
@@ -154,32 +165,35 @@ func (hdb *HostDB) threadedScan() {
 			}
 
 			// Select the first InactiveHostCheckupQuantity hosts from the
-			// shuffled list.
+			// shuffled list and scan them.
 			n := InactiveHostCheckupQuantity
 			if len(random) < InactiveHostCheckupQuantity {
 				n = len(random)
 			}
 			for i := 0; i < n; i++ {
-				go hdb.threadedProbeHost(random[i])
+				go func() {
+					hdb.scanPool <- random[i]
+				}()
```

Comment:
this appears often enough that it might be cleaner to separate it into a `hdb.addToScanPool` function. Then you wouldn't need to explain about blocking channels and such.